pilot/pkg/proxy/envoy/discovery.go

```diff
@@ -472,8 +499,11 @@ func (ds *DiscoveryService) clearCache() {
 			time.Since(lastClearCacheEvent), time.Since(lastClearCache))
 		lastClearCacheEvent = time.Now()
 		lastClearCache = time.Now()
-
-		V2ClearCache()
+		full := ds.fullPush
+		svc := ds.edsUpdatedServices
+		ds.fullPush = false
+		ds.edsUpdatedServices = []string{}
+		Push(full, svc)
```

Comment:
Yes, I thought about that and may do it - the alternative is to remove the 2 redundant calls, which are currently not used ( it was a fail safe in case we ran into issues with debouncing ).